### PR TITLE
test compact reader with default methods

### PR DESCRIPTION
**Pull request by sancar**

### CODE REVIEW

File: hazelcast/src/test/java/example/serialization/InnerDTO.java [hunk 1]
```diff
@@ -143,23 +134,20 @@ public int hashCode() {
...
-                + ", + ll=" + Arrays.toString(ll)
-                + ", + ff=" + Arrays.toString(ff)
-                + ", + dd=" + Arrays.toString(dd)
+                + "+ bb=" + Arrays.toString(bytes)
+                + ", + ss=" + Arrays.toString(shorts)
+                + ", + ii=" + Arrays.toString(ints)
+                + ", + ll=" + Arrays.toString(longs)
+                + ", + ff=" + Arrays.toString(floats)
+                + ", + dd=" + Arrays.toString(doubles)
+                + ", + dd=" + Arrays.toString(strings)
```
Comment (utku-caglayan) [inline hazelcast/src/test/java/example/serialization/InnerDTO.java:150]: "dd=" creates a conflict with doubles. Can we change this?

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java [hunk 2]
```diff
@@ -59,12 +59,12 @@ static GenericRecord createCompactGenericRecord(MainDTO mainDTO) {
...
-                .setArrayOfInt32("i", inner.ii)
-                .setArrayOfInt64("l", inner.ll)
-                .setArrayOfFloat32("f", inner.ff)
-                .setArrayOfFloat64("d", inner.dd)
+                .setArrayOfInt8("b", inner.bytes)
+                .setArrayOfInt16("s", inner.shorts)
+                .setArrayOfInt32("i", inner.ints)
+                .setArrayOfInt64("l", inner.longs)
+                .setArrayOfFloat32("f", inner.floats)
+                .setArrayOfFloat64("d", inner.doubles)
```
Comment (utku-caglayan) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java:67]: We should also set inner.strings

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java [hunk 3]
```diff
@@ -72,6 +83,85 @@ public void testAllTypesWithReflectiveSerializer() {
...
+        assertArrayEquals(new Byte[0], innerDTO.nullableBytes);
+        assertArrayEquals(new Short[0], innerDTO.nullableShorts);
+        assertArrayEquals(new Integer[0], innerDTO.nullableIntegers);
+        assertArrayEquals(new Long[0], innerDTO.nullableLongs);
+        assertArrayEquals(new Float[0], innerDTO.nullableFloats);
+        assertArrayEquals(new Double[0], innerDTO.nullableDoubles);
+    }
+
+    @Test
+    public void testReaderReturnsDefaultValues_whenDataIsMissing() {
```
Comment (mdumandag) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java:148]: are the test method names mixed? I see that we are reading missing data in the other method. here, the main dto is fully constructed with actual data

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java (hunk 2 above)
Comment (utku-caglayan) [reply hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java:67]: Also I think one of the tests should fail if this is the case 😅, should we add such a test?

**Review by utku-caglayan — COMMENTED** [review]
Few comments

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java (hunk 3 above)
Comment (sancar) [reply hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactStreamSerializerTest.java:148]: Yes, copy paste error. Fixing it.

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java (hunk 2 above)
Comment (sancar) [reply hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactTestUtil.java:67]: This is a utility method to something else with GenericRecords. I think indication of this is missing should be code coverage not a failing test. I am adding the missing call to setArrayOfStrings

**Comment by sancar** [comment]
run-lab-run

**Comment by sancar** [comment]
run-lab-run

**Review by utku-caglayan — APPROVED** [review]
Looks great!

**Review by mdumandag — APPROVED** [review]

**Pull request merged**